lib/contracts/deploy_manager.js

```diff
@@ -1,22 +1,54 @@
 let async = require('async');
-//require("../utils/debug_util.js")(__filename, async);
-let Deploy = require('./deploy.js');
 
 class DeployManager {
   constructor(options) {
+    const self = this;
     this.config = options.config;
     this.logger = options.logger;
     this.blockchainConfig = this.config.blockchainConfig;
 
     this.events = options.events;
     this.plugins = options.plugins;
     this.blockchain = options.blockchain;
-    this.chainConfig = (options.trackContracts !== false) ? this.config.chainTracker : false;
-    this.contractsManager = options.contractsManager;
     this.gasLimit = false;
     this.fatalErrors = false;
     this.deployOnlyOnConfig = false;
     this.onlyCompile = options.onlyCompile !== undefined ? options.onlyCompile : false;
+
+    this.events.setCommandHandler('deploy:contracts', (cb) => {
+      self.deployContracts(cb);
+    });
+  }
+
+  deployAll(done) {
+    let self = this;
+    this.logger.info(__("deploying contracts"));
+    this.events.emit("deploy:beforeAll");
+
+    self.events.request('contracts:list', (contracts) => {
+      async.eachOfSeries(contracts,
+        function (contract, key, callback) {
+          contract._gasLimit = self.gasLimit;
+          self.events.request('deploy:contract', contract, () => {
+            callback();
+          });
+        },
+        function (err, _results) {
+          if (err) {
+            self.logger.error(__("error deploying contracts"));
+            self.logger.error(err.message);
+            self.logger.debug(err.stack);
```

Comment:
Shouldn't you callback here with the error? That way we don't log "Finished deploying".

Reply:
perhaps, for now I just 'moved' the code here from deploy.js and kept the same behaviour, while keep this in mind when refactoring